lang/wasm.cc

```diff
@@ -264,7 +275,7 @@ struct from_val_visitor {
 
         if (get_abi(instance, store, mem_base) == 2) {
             auto free_func = import_func(instance, store, "_scylla_free");
-            call_func(store, free_func, {val.i32()});
+            call_void_func(store, free_func, {wasmtime::Val((int32_t)val.i64())});
         }
```

Comment:
Is this unrelated to the cache? If so move it to a separate pull request.

Reply:
Ok